holmes/core/toolset_manager.py

```diff
@@ -0,0 +1,394 @@
+            logging.info("refreshing toolset status")
+            self.refresh_toolset_status(dal)
+
+        cached_toolsets: List[dict[str, Any]] = []
+        with open(self.toolset_status_location, "r") as f:
+            cached_toolsets = json.load(f)
+
+        # load status from cached file and update the toolset details
+        toolsets_status_by_name: dict[str, dict[str, Any]] = {
+            cached_toolset["name"]: cached_toolset for cached_toolset in cached_toolsets
+        }
+        all_toolsets_with_status = self._list_all_toolsets(
+            dal=dal, check_prerequisites=False
+        )
+
+        for toolset in all_toolsets_with_status:
+            if toolset.name in toolsets_status_by_name:
+                # Update the status and error from the cached status
+                cached_status = toolsets_status_by_name[toolset.name]
+                toolset.status = ToolsetStatusEnum(cached_status["status"])
+                toolset.error = cached_status.get("error", None)
+                toolset.enabled = cached_status.get("enabled", True)
+                toolset.type = ToolsetType(
+                    cached_status.get("type", ToolsetType.BUILTIN)
+                )
+                toolset.path = cached_status.get("path", None)
+
+        # CLI custom toolsets status are not cached, and their prerequisites are always checked whenever the CLI runs.
+        custom_toolsets_from_cli = self._load_toolsets_from_paths(
+            self.custom_toolsets_from_cli,
+            list(toolsets_status_by_name.keys()),
+            check_conflict_default=True,
+        )
+        # custom toolsets from cli as experimental toolset should not override custom toolsets from config
+        for custom_toolset_from_cli in custom_toolsets_from_cli:
+            if custom_toolset_from_cli in toolsets_status_by_name.keys():
+                raise Exception(
+                    f"Toolset {custom_toolset_from_cli.name} from cli is already defined in existing toolset"
+                )
+        all_toolsets_with_status.extend(custom_toolsets_from_cli)
+
+        return all_toolsets_with_status
+
+    def list_enabled_console_toolsets(
+        self, dal: Optional[SupabaseDal] = None, refresh_status=False
+    ) -> List[Toolset]:
+        """
+        List all enabled toolsets that cli tools can use.
+
+        listing console toolset does not refresh toolset status by default, and expects the status to be
+        refreshed specifically and cached locally.
+        """
+        toolsets_with_status = self.load_toolset_with_status(
+            dal, refresh_status=refresh_status
+        )
+        return [
+            ts
+            for ts in toolsets_with_status
+            if any(tag in self.cli_tool_tags for tag in ts.tags)
+        ]
+
+    # TODO(mainred): cache and refresh periodically toolset status for server if necessary
+    def list_enabled_server_toolsets(
```

Comment:
Looks like for some reason when running with this version on my local cluster the datetime toolset isn't being enabled by default.